_projects/coala_language_server.md

```diff
@@ -0,0 +1,63 @@
+---
+collaborating_projects:
+  - coala
+desc: "Offer coala languager server."
+difficulty: medium
+initiatives:
+  - GSoC
+  - RGSoC
+issues:
+  - "https://github.com/coala/coala-vs-code/issues/3"
+  - "https://github.com/coala/coala-vs-code/issues/4"
+markdown: coala_language_server.md
+mentors:
+  - gaocegege
+name: "coala Language Server"
+requirements:
+  - "The applicant should have at least one patch accepted to any of the coala."
+  - "The applicant shoule be familiar with language server protocol"
+tags:
+  - Language Server
+  - Plugins
+---
+
+The [Language Server Protocol](https://langserver.org/) was created by Microsoft to define a common language for programming language analyzers to speak. Today, several companies have come together to support its growth, including Codenvy, Red Hat, and Sourcegraph, and the protocol is becoming supported by a rapidly growing list of editor and language communities:
+
+- VS Code
+- Eclipse Che
+- Eclipse IDE(Eclipse LSP4E)
+- IntelliJ / JetBrains IDEs
+- Vim
+- Atom (WIP)
+- Emacs (WIP)
+- Sublime (WIP)
+
+We have a state-of-art implementation in [coala-vs-code](https://github.com/coala/coala-vs-code) but it is designed for VS Code only. We need to design and implement a general mechanism for multiple editors/IDEs support.
+
+The project is divided into two parts:
+
+1. Offer a basic languager server.
+2. Implement a plugin based on the language server.
+
+#### Milestones
+
+##### PREPARATION/BONDING
+
+- An official repository exists with CI.
+- An mock language server exists.
+
+##### CODING PHASE 1
+
+- A basic language server for coala exists.
```

Comment:
which features are 'basic'?

Reply:
I think it depends on the students. For me, linting feature is enough.

Reply:
I suspect that this project might only do linting for the entire project.

Integrating other parts of LS, like symbol lookup, etc, would only result in more copying of the other Python LS, which use jedi or similar.

coala isnt capable of providing the other parts of LS, at least not yet.

Perhaps a better approach is to build a proxy LS, which allows parts of LSP to be implemented in one LS, and other parts of LSP to be served by another LSP, with the proxy choosing which LSP to use.

Reply:
SGTM